### DISCUSSION TITLE
Error on package that is NPM linked

### DISCUSSION THREAD

**Discussion by jasonericdavis** [remix-run/remix#2831] [General]
Hello,

While trying to use a package that is npm linked  an error is thrown within the server-bare-modules plugin.

### Steps to reproduce:
1. In a folder outside of the remix project create a npm project using the `npm init` command. Start the package name with the `@` characer.
2. Within the folder of the folder created in step 1, Run the `npm link` command  create a local global package.
3. In your remix project:
     -  Run the command  `npm link <name of package from step 2>` to link the two projects.
    -   import the package in a route file and use it. Note just importing the package will not cause the problem. 
4. Try to build the remix project.

### Expected Outcome:
- The project builds successfully

### Actual Outcome:
- An Error is thrown

### Notes
The npm link command creates a symlink to the local package which cause the function [resolveModuleBasePath(packageName: string, fullImportPath: string) ](https://github.com/remix-run/remix/blob/a2e1449760f788cc424421e53eb98ae82d4f64d1/packages/remix-dev/compiler/plugins/serverBareModulesPlugin.ts#L174) in the serverBareModules plugin to pull the full path of the imported package instead of the path to the node_modules folder when it doing the `require.resolve()`

This error won't occur if the name of the package is the same as the linked folder.

### Work Around
The current work around is to update the import to pull from the node_modules folder as a relative path. 

https://github.com/remix-run/remix/blob/a2e1449760f788cc424421e53eb98ae82d4f64d1/packages/remix-dev/compiler/plugins/serverBareModulesPlugin.ts#L174

**Comment by Plopix** [comment]
I confirm the bug and the workaround ;-) I also confirm that the same package not symlinked is working.
Linking package for local development and building libraries is important and a valid use case for this to be fix I believe! Thanks!

**Comment by tonydotgit** [comment]
I'm facing the same issue right now
